larryTheGeek/RideApi
# API end points That pass Test

**What does this PR do?**
Refactors the code to pass the test by adding API end points

**Endpoints added**

EndPoint | Functionality
-- | --
POST /api/v1/rides | Creates a new ride
GET /api/v1/rides | Fetches all available rides
GET /api/v1/rides/<rideId> | Fetches a single ride
POST /api/v1/rides/<rideId>/requests | Creates a ride offer
POST/api/v1/register| Registers a new user
GET  /api/v1/users| Gets all the registered users
POST/api/v1/login| logs in a registered user

**Task achieved in this PR**
- create API end to get all available ride offers
- create API end point to get a specific ride offer
- create API end point to create a ride offer
- create API  to request to join a ride offer
- create API end point to login and authenticate a user
- create API end point to add a new user
- create API end point to get all users

**How should this be manually tested?**

Clone this repo using git clone 
Check the ReadMe for instructions on setting up the project.
git checkout f1
test the above endpoints using post-man

**Relevant Pivotal stories**
https://www.pivotaltracker.com/n/projects/2180294

![ure](https://user-images.githubusercontent.com/22986772/41918043-3116256c-7964-11e8-8bda-b02289038c0d.PNG)
